py/modmicropython.c

```diff
@@ -130,9 +130,17 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_0(mp_micropython_heap_lock_obj, mp_micropython_he
 
 STATIC mp_obj_t mp_micropython_heap_unlock(void) {
     gc_unlock();
-    return mp_const_none;
+    return MP_OBJ_NEW_SMALL_INT(MP_STATE_MEM(gc_lock_depth));
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_0(mp_micropython_heap_unlock_obj, mp_micropython_heap_unlock);
+
+#if MICROPY_PY_ENABLE_HEAP_LOCKED
```

Comment:
Sorry, forgot to say that all these new macro configs should be changed to `MICROPY_PY_MICROPYTHON_HEAP_LOCKED`.

Reply:
no that's absolutely my fault for rushing :-S